### PR TITLE
Cleanup nx colors RSC-496

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-496

Moved unused and old colour variables into a deprecated section at the bottom of `_colors.scss`.

Looked through the existing styles in RSC and where we had hex values that could be replaced with variables I did so. Where the colours were unique I created tickets assigned to @dturner32 to come up with colour options within the RSC palettes.

### CODE REVIEW

File: lib/src/scss-shared/_nx-color-swatches.scss [hunk 1]
```diff
@@ -111,6 +111,7 @@ $nx-red-300: #ff8286;
 $nx-red-200: #ff979a;
 $nx-red-100: #ffb6b8;
 $nx-red-50: #ffcbcc;
+$nx-red-25: #fff6f8;
```
Comment (rpokorny) [inline lib/src/scss-shared/_nx-color-swatches.scss:114]: Shouldn't this be checked with @dturner32 ?

File: lib/src/scss-shared/_nx-colors.scss [hunk 2]
```diff
@@ -131,3 +76,66 @@ $nx-selectable-colors: (
   yellow: ($nx-yellow-700, $nx-yellow-50),
   lime: ($nx-green-200, $nx-green-50)
 );
+
+/*
+  WARNING
```
Comment (rpokorny) [inline lib/src/scss-shared/_nx-colors.scss:81]: I think this section would be a lot better as a separate file than a "section" at the bottom of this file

File: lib/src/scss-shared/_nx-colors.scss [hunk 3]
```diff
@@ -131,3 +76,66 @@ $nx-selectable-colors: (
...
+
+$nx-fail-color-active: #5c0000;
+$nx-fail-color-disabled: #dd9595;
+$nx-fail-color-hover: #5d0000;
+
+// Border rows are used by lists and tables
+$nx-border-row: 1px solid $nx-grey-1;
+$nx-border-row-hover: 1px solid $nx-grey-4;
+$nx-border-row-disabled: 1px solid $nx-grey-1;
+$nx-border-row-dark: 1px solid $nx-grey-5;
```
Comment (rpokorny) [inline lib/src/scss-shared/_nx-colors.scss:141]: These aren't colors and so I would think should not be in this section.

File: lib/src/scss-shared/_nx-color-swatches.scss (hunk 1 above)
Comment (glenhunter) [reply lib/src/scss-shared/_nx-color-swatches.scss:114]: > Shouldn't this be checked with @dturner32 ?

I confirmed all the colour changes with @dturner32. He came up with the colour above.

File: lib/src/scss-shared/_nx-colors.scss [hunk 4]
```diff
@@ -94,18 +46,12 @@ $nx-disabled-border: $nx-grey-500;
...
-$nx-border-row: 1px solid $nx-grey-1;
-$nx-border-row-hover: 1px solid $nx-grey-4;
-$nx-border-row-disabled: 1px solid $nx-grey-1;
-$nx-border-row-dark: 1px solid $nx-grey-5;
+$nx-border-darker: 1px solid $nx-grey-800;
+$nx-border-dark: 1px solid $nx-grey-300;
+$nx-border: 1px solid $nx-grey-200; // default
+$nx-border-light: 1px solid $nx-grey-100;
+$nx-border-lighter: 1px solid $nx-grey-50;
+$nx-border-error: 1px solid $nx-red-800;
```
Comment (rpokorny) [inline lib/src/scss-shared/_nx-colors.scss:54]: Should these also be moved to `_nx-variables` while we're at it?

**Review by rpokorny — APPROVED** [review]

**Pull request merged**